### ISSUE TITLE
NextJS: getSession() is always returning null in api route

### ISSUE THREAD

**Issue by Neo-Ciber94** [auth0/nextjs-auth0#575]
### Description

I have an API route but when trying to access the session using ``getSession()`` the result is always null.

### Reproduction

Create a new NextJS App.

App the following routes:
```ts
/// pages/api/[[...auth0]].ts

import { handleAuth } from "@auth0/nextjs-auth0";

export default handleAuth();
```

```ts
/// pages/api/data.ts

import { getSession } from "@auth0/nextjs-auth0";
import { NextApiRequest, NextApiResponse } from "next";

export default function handler(req: NextApiRequest, res: NextApiResponse) {
  const session = getSession(req, res); // This is always null
  console.log("SESSION: ", { session });
  return res.send({ data: "Hello World" });
}
```
```ts
/// pages/index.tsx

import type { GetServerSideProps, InferGetServerSidePropsType } from "next";

type Data = {
  data: string;
};

export const getServerSideProps: GetServerSideProps<Data> = async () => {
  const res = await fetch("http://localhost:3000/api/data");
  const data = await res.json();
  return { props: { data } };
};

const Home = ({ data }: InferGetServerSidePropsType<typeof getServerSideProps>) => {
  let text = JSON.stringify(data, null, 2);
  return <pre style={{ padding: 20 }}>{text}</pre>;
};

export default Home;
```

This asume you have a ```.env.local``` setup with auth0, then go to ``http://localhost:3000/api/auth/login`` and login normally but after going back to ``http://localhost:3000`` the console is logging:

```bash
SESSION:  { session: null }
```

If the api route is changed to:

```ts
/// pages/api/data.ts
import { getSession, withApiAuthRequired } from "@auth0/nextjs-auth0";

export default withApiAuthRequired(function (req, res) {
  const session = getSession(req, res);
  console.log("SESSION: ", { session });
  return res.send({ data: "Hello World" });
});
```

In the browser ``http://localhost:3000`` is showing:

```json
{
  "error": "not_authenticated",
  "description": "The user does not have an active session or is not authenticated"
}
```

But when using ``getSession()`` in ``getServerSideProps`` or during the ``callback`` the session is returned corretly.

### Environment

-  ``"@auth0/nextjs-auth0": "^1.7.0"``
- ``"next": "12.0.7",``
- Google Chrome Versión 97.0.4692.71

**Comment by adamjmcgrath** [comment]
Hi @Neo-Ciber94 

Have a look at https://github.com/auth0/nextjs-auth0/issues/534#issuecomment-965584244 

tl;dr you shouldn't call `fetch("http://localhost:3000/api/data")` from `getServerSideProps`

**Comment by Neo-Ciber94** [comment]
Thanks that was my problem, using the cookies from the request solved it.

```ts
export const getServerSideProps = withPageAuthRequired<Data>({
  getServerSideProps: async ({ req }) => {
    const response = await fetch("http://localhost:3000/api/data", {
      headers: {
        cookie: req.headers.cookie || "",
      },
    });
    const data = await response.json();
    return { props: { data } };
  },
});
```

**Comment by babysteps** [comment]
@adamjmcgrath i am having a similar issue, however I am not calling the API route that contains getSession() from getServerSideProps.  i am using useSWR w/ axios to make the call from the client.  on my local, getSession() returns the session object, however, on Vercel, it always returns null, and clears the appSession.0 and appSession.1 cookies.  when i was using 2.0.0-beta.2 it was working fine, but since upgrading to 2.2.1 i get the null issue.  any ideas?

**Comment by lihail** [comment]
> @adamjmcgrath i am having a similar issue, however I am not calling the API route that contains getSession() from getServerSideProps. i am using useSWR w/ axios to make the call from the client. on my local, getSession() returns the session object, however, on Vercel, it always returns null, and clears the appSession.0 and appSession.1 cookies. when i was using 2.0.0-beta.2 it was working fine, but since upgrading to 2.2.1 i get the null issue. any ideas?
> 
> also, i logged out req.cookies right BEFORE getSession() and they were there. but after, they're gone.

@babysteps did you happen to find a solution for this?
